### PR TITLE
Turbo Composer logic_gate addition

### PR DESCRIPTION
**Pull request by CPerezz**
Included Logic constraint gate for the `TurboComposer`.

Instead of exposing it directly, I preferred to do that:
```rust
/// Adds a logical XOR gate that performs the XOR between two values for the
    /// specified first `num_bits` returning a `Variable` holding the result.
    ///
    /// # Panics
    ///
    /// If the `num_bits` specified in the fn params is odd.
    pub fn logic_xor_gate(&mut self, a: Variable, b: Variable, num_bits: usize) -> Variable {
        self.logic_gate(a, b, num_bits, true)
    }

    /// Adds a logical AND gate that performs the bitwise AND between two values
    /// for the specified first `num_bits` returning a `Variable` holding the result.
    ///
    /// # Panics
    ///
    /// If the `num_bits` specified in the fn params is odd.
    pub fn logic_and_gate(&mut self, a: Variable, b: Variable, num_bits: usize) -> Variable {
        self.logic_gate(a, b, num_bits, false)
    }
```

### CODE REVIEW

File: src/constraint_system/standard/quotient_poly.rs [hunk 1]
```diff
@@ -156,7 +159,9 @@ fn compute_circuit_satisfiability_equation(
             let b_4 = delta(w4_next - four * wl);
             let b = (b_1 + b_2 + b_3 + b_4) * qrange;
 
-            (a + b) * v_h_i
+            let c = qlogic * ((wl - wr) * wo);
```
Comment (kevaundray) [inline src/constraint_system/standard/quotient_poly.rs]: The check for the logic gates cannot be just `(wl - wr) * wo)`

Comment (CPerezz) [reply src/constraint_system/standard/quotient_poly.rs]: Agree as we discussed.

Actually missing:
- XOR/AND constraint
- Base4 (quad) constraint (For the input wires and output)

**Comment by CPerezz** [comment]
I need to change the quad_iterator for the bititerator ported from zcash repo by @kevaundray .
Then would be LGTM

**Comment by CPerezz** [comment]
@kevaundray all ready. 

- Switch to `BitIterator` done. 
- Updated tests.
- Added more docs & notes on the code

Feel free to merge!

**Pull request merged**